editoast/src/views/search.rs

```diff
@@ -333,10 +337,19 @@ struct SearchDBResult {
     )
 )]
 async fn search(
-    Query(query_params): Query<PaginationQueryParam>,
     State(db_pool): State<DbConnectionPoolV2>,
+    Extension(authorizer): AuthorizerExt,
+    Query(query_params): Query<PaginationQueryParam>,
     Json(SearchPayload { object, query, dry }): Json<SearchPayload>,
 ) -> Result<impl IntoResponse> {
+    let authorized = authorizer
+        .check_roles([BuiltinRole::InfraRead].into())
+        .await
+        .map_err(AuthorizationError::AuthError)?;
+    if !authorized {
+        return Err(AuthorizationError::Unauthorized.into());
+    }
+
```

Comment:
It's more complicated than that, the search engine is dynamic and isn't restricted to infra object search. Use `object` to determine the correct roles to check.

Reply:
as discussed, we will use compile-time defined strings to check roles based on the `object` field in the `SearchPayload`.